decentraland/unity-explorer
# Chore: default memory cap and settings for it

## What does this PR change?
Close #2242

- added possibility to change memory limit from the settings menu. 
- added persistent saving of memory settings
- forced full memory usage on the first loading flow (initial loading screen)

## How to test the changes?

1. Launch the explorer
2. .Open /debug -> memory and observe that Memory budget threshold is ~10_000/~12_000 (which is 10 GB/ 12 GB)
3. Open Menu settings graphics -> observe new dorpdown for Memory Limit
4. Choose different ones and see how it reflected in the Debug menu. Note, that the red value will be 75% of the chosen value (also converted from GB to MB)
5. Choose Max and observe that red value is 75% of your Max. For example, for my 64 GB it was ~48 GB shown
6. Close and open the app.
7. Verify in dropdown that selected value was saved

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
